lib/Runtime/Library/DateImplementation.cpp

```diff
@@ -1073,7 +1073,22 @@ namespace Js {
         size_t size = sizeof(char16) * (ulength + 1);
         js_memcpy_s(pszSrc, size, psz, size);
 
+        #ifdef _WIN32
         _wcslwr_s(pszSrc,ulength+1);
+        #else
+        //fast xplat conversion from upper case to lower case
+        //only targets the UTF16 Latin alphabet - but that's all we need here
+        //used instead of the PAL _wcslwr_s which takes ~100x as long to run
+        int i;
+        for (i = 0 ; pszSrc[i] != 0; i++)
```

Comment:
We actually have a length here, I'd prefer it were used rather than the null terminator.